stock_release_channel_partner_delivery_window/models/stock_release_channel.py

```diff
@@ -17,26 +18,23 @@ class StockReleaseChannel(models.Model):
     )
 
     def filter_release_channel_partner_window(self, picking, partner):
-        channels = self
-        if (
-            not partner.delivery_time_preference
-            or partner.delivery_time_preference == "anytime"
-        ):
-            return channels
-
-        for channel in self:
-            if not channel.shipment_date:
-                continue
-            shipment_datetime = fields.Datetime.to_datetime(channel.shipment_date)
-            if channel.process_end_date:
-                shipment_datetime = shipment_datetime.replace(
-                    hour=channel.process_end_date.hour,
-                    minute=channel.process_end_date.minute,
+        if picking.sale_id.commitment_date:
+            # Date is forced on SO, don't filter
+            return self
+        if partner.delivery_time_preference == "anytime":
+            return self
+        if partner.delivery_time_preference == "workdays":
+            open_days = set(range(5))
+        else:
+            open_days = set(
+                map(
+                    int,
+                    partner.delivery_time_window_ids.time_window_weekday_ids.mapped(
+                        "name"
+                    ),
```

Comment:
I guess we would benefit from a method on the mixin to return open days.
```suggestion
        if partner.delivery_time_preference == "anytime":
            return self
        open_days = partner._get_open_days()
```
This would be kinda useful, as I suppose open days might vary depending on the country maybe.

Reply:
good idea
https://github.com/OCA/stock-logistics-workflow/pull/1940